val foo by <!DEBUG_INFO_MISSING_UNRESOLVED, TYPECHECKER_HAS_RUN_INTO_RECURSIVE_PROBLEM!>foo<!>

val bar by <!DELEGATE_SPECIAL_FUNCTION_MISSING!>baz(<!DEBUG_INFO_MISSING_UNRESOLVED, TYPECHECKER_HAS_RUN_INTO_RECURSIVE_PROBLEM!>bar<!>)<!>

fun <T> baz(t: T): T = t
